src/Ldap.php

```diff
@@ -850,11 +959,45 @@ public function bind($username = null, $password = null)
 
             $zle = new Exception\LdapException($this, $message);
         }
-        $this->disconnect();
+        $this->unbind();
 
         throw $zle;
     }
 
+    protected function shouldReconnect($resource)
+    {
+        if ($this->reconnectCount < $this->getReconnectsToAttempt()
+            && ldap_errno($resource) === -1
+        ) {
+            $this->reconnectCount++;
+            $this->reconnectSleep();
+
+            try {
+                $this->connect();
+                $this->bind();
+
+                $this->reconnectsAttempted = $this->reconnectCount;
+                $this->reconnectCount = 0;
+                return true;
+            } catch (LdapException $e) {
+                if ($e->getCode() === -1) {
```

Comment:
What happens when there is an LDAPException with a code different to ```-1```? we silently do nothing? Or should we rethrow the Exception? Alternatively perhaps throwing a new Exception would be an option as well…

Reply:
Good question. As it's written I think the code "Does The Right Thing," but it is not very self-documenting. If you look at everywhere `Ldap::shouldReconnect()` is used, there was already code immediately following whatever `ldap_*()` function call that checks the return value and converts failures expressed in return values to exceptions.